pennylane/templates/subroutines/out_adder.py

```diff
@@ -297,23 +299,33 @@ def _out_adder_decomposition_resources(num_output_wires, num_x_wires, num_y_wire
     else:
         qft_wires = num_output_wires
 
-    resources = Counter(
-        {
-            resource_rep(QFT, num_wires=qft_wires): 1,
-            adjoint_resource_rep(QFT, {"num_wires": qft_wires}): 1,
-        }
-    )
-
-    for num_wires in (num_x_wires, num_y_wires):
-        rep = resource_rep(
+    target_resources = defaultdict(int)
+    target_resources[
+        resource_rep(
             ControlledSequence,
             base_class=PhaseAdder,
             base_params={"num_x_wires": qft_wires, "mod": mod},
-            num_control_wires=num_wires,
+            num_control_wires=num_x_wires,
         )
-        resources[rep] += 1
+    ] += 1
+    target_resources[
+        resource_rep(
+            ControlledSequence,
+            base_class=PhaseAdder,
+            base_params={"num_x_wires": qft_wires, "mod": mod},
+            num_control_wires=num_y_wires,
+        )
+    ] += 1
 
-    return dict(resources)
+    params = {
+        "compute_op_params": {"num_wires": qft_wires},
+        "target_op_params": {
+            "resources": dict(target_resources),
+        },
+        "uncompute_op_params": {"num_wires": qft_wires},
```

Comment:
This suggestion depends a bit on what you decide to do for the `change_op_basis_resource_rep` behaviour with `uncompute_op=None`.
```suggestion
```